doc/2/api/controllers/document/delete-fields/index.md

```diff
@@ -0,0 +1,95 @@
+---
+code: true
+type: page
+title: delete-fields
+description: Delete fields of an existing document.
+---
+
+# deleteFields
+
+Delete fields of an existing document.
+
+---
+
+## Query Syntax
+
+### HTTP
+
+```http
+URL: http://kuzzle:7512/<index>/<collection>/<documentId>/_fields[?refresh=wait_for]
+Method: DELETE
+Body:
+```
+
+```js
+{
+  "fields": [
+    // path of fields to remove
+  ]
+}
+```
+
+### Other protocols
+
+```js
+{
+  "index": "<index>",
+  "collection": "<collection>",
+  "controller": "document",
+  "action": "deleteFields",
+  "_id": "<documentId>",
+  "refresh": "wait_for",
+  "body": {
+    "fields": [
+      // path of fields to remove
+    ]
+  }
+}
+```
+
+---
+
+## Arguments
+
+- `collection`: collection name
+- `documentId`: unique ID of the document where the fields should be removed
+- `index`: index name
+
+### Optional:
+
+- `refresh`: if set to `wait_for`, Kuzzle will not respond until the new document content is indexed
+
+---
+
+## Body properties
+
+- `fields`: an array of string. Each string represents a path (see [lodash path](https://lodash.com/docs/4.17.15#toPath)) to a specific field to remove
+
+---
+
+## Response
+
+Returns an object containing updated document information, with the following properties:
+
+- `_id`: document unique identifier
+- `_source`: new document content
```

Comment:
Maybe we could add a `source` option to this method like for the `document:update` method so Kuzzle will return the body only if asked.

What do you think?

Reply:
This could reduce the network load is the user doesn't need to get the new document, I will add an option for that